netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1521,6 +1445,11 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
         goto Exit;
     }
 
+    // Populate the sock_addr context with WFP classify input fields.
```

Comment:
what was the need to move this up?

Reply:
This was a bug. This function is accessing sock_addr struct at many places even before it was initialized (but it was harmless till now as it is only accessing the compartment id from the struct). If we attach multiple programs to different compartments, we start hitting an assert in the code

Reply:
Please file test gap issue.

Reply:
Already filed #2574